core/ecschema-metadata/src/Metadata/Constant.ts

```diff
@@ -27,6 +27,8 @@ export class Constant extends SchemaItem {
   protected _definition: string;
   protected _numerator: number;
   protected _denominator: number;
+  protected _isNumeratorExplicitlyDefined: boolean = false;
```

Comment:
Why add two new properties instead of making `_numerator` and `_denominator` of type `number | undefined`?

Reply:
One thing to consider is how we handle default values in C++ vs TypeScript.  In C++ we set the property to the default value if it is not set in the xml/json.  In TypeScript we are leaving the values unset, this means we do not need separate flags to tell us if the properties are set but consuming code does need to be able to handle unset values.  

I think we should continue leaving the values unset in TypeScript code.  So we do not need the flags

Reply:
ok let me do that. I was following the c++ code earlier.

Reply:
I have removed the new properties.